ugvc/mrd/mrd_utils.py

```diff
@@ -476,40 +477,47 @@ def intersect_featuremap_with_signature(
         raise OSError(f"Output file {output_intersection_file} already exists and overwrite flag set to False")
     # make sure vcf is indexed
     signature_file = _safe_tabix_index(signature_file)
-    # build a set of all signature entries, including alts and ref
-    signature_entries = set()
-    with pysam.VariantFile(signature_file) as f_sig:
-        for rec in f_sig:
-            signature_entries.add((rec.chrom, rec.pos, rec.ref, rec.alts))
-    # Only write entries from featuremap to intersection file if they appear in the signature with the same ref&alts
-    try:
-        with pysam.VariantFile(featuremap_file) as f_feat:
-            header = f_feat.header
-            if add_info_to_header is not None:
-                header.add_line(
-                    "##File:Description=This file is an intersection of a featuremap with a somatic mutation signature"
-                )
-                header.add_line(f"##python_cmd:intersect_featuremap_with_signature=python {' '.join(sys.argv)}")
-                if is_matched is not None:
-                    if is_matched:
-                        header.add_line("##Intersection:type=matched (signature and featuremap from the same patient)")
-                    else:
-                        header.add_line(
-                            "##Intersection:type=control (signature and featuremap not from the same patient)"
-                        )
-            with pysam.VariantFile(output_intersection_file_vcf + ".tmp", "w", header=header) as f_int:
-                for rec in f_feat:
-                    if ((not complement) and ((rec.chrom, rec.pos, rec.ref, rec.alts) in signature_entries)) or (
-                        complement and ((rec.chrom, rec.pos, rec.ref, rec.alts) not in signature_entries)
-                    ):
-                        f_int.write(rec)
-        os.rename(output_intersection_file_vcf + ".tmp", output_intersection_file_vcf)
-    finally:
-        if "output_intersection_file_vcf" in locals() and os.path.isfile(output_intersection_file_vcf + ".tmp"):
-            os.remove(output_intersection_file_vcf + ".tmp")
-    # index output
-    pysam.tabix_index(output_intersection_file_vcf, preset="vcf", force=True)  # this also compressed the vcf to vcf.gz
-    assert os.path.isfile(output_intersection_file_vcf + ".gz")
+    
+    # Use bcftools isec to intersect the two VCF files by position and compress the output
+    cmd = f"bcftools isec -n=2 -w1 -Oz -o isec.vcf.gz {signature_file} {featuremap_file}"
```

Comment:
You shouldn't be writing files into the current working directory, you could overwrite something. You have "output_intersection_file", you can write temporary files to the same directory. I like doing the following:

```suggestion
    with tempfile.TemporaryDirectory(
        dir=os.path.dirname(output_intersection_file)
    ) as temp_dir:
        isec_file = os.path.join(temp_dir, "isec.vcf.gz")
        header_file = os.path.join(temp_dir, "header.txt")
        headerless_featuremap = os.path.join(temp_dir, "headerless_featuremap.vcf.gz")
```
And then use these. You don't even have to delete the files when you are done, they are deleted when you leave this "with" clause